src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.13 on 2019-12-30 20:52
+from __future__ import unicode_literals
+
+import os
+from datetime import timedelta, datetime
+
+from django.db import migrations
+
+from sentry import options
+
+
+def backfill_eventstream(apps, schema_editor):
+    """
+    Inserts Postgres events into the eventstream if there are recent events in Postgres.
+
+    This is for open source users migrating from 9.x who want to keep their events.
+    If there are no recent events in Postgres, skip the backfill.
+    """
+    from sentry import eventstore, eventstream
+    from sentry.utils.query import RangeQuerySetWrapper
+
+    Event = apps.get_model('sentry', 'Event')
+    Group = apps.get_model('sentry', 'Group')
+    Project = apps.get_model('sentry', 'Project')
+
+    # Kill switch to skip this migration
+    skip_backfill = os.environ.get("SKIP_EVENTS_BACKFILL", False)
+
+    # Use 90 day retention if the option has not been set or set to 0
+    DEFAULT_RETENTION = 90
+    retention_days = options.get("system.event-retention-days") or DEFAULT_RETENTION
+
+    def get_events(last_days):
+        to_date = datetime.now()
+        from_date = to_date - timedelta(days=last_days)
+        return Event.objects.filter(datetime__gte=from_date, datetime__lte=to_date)
+
+    def _attach_related(_events):
+        project_ids = {event.project_id for event in _events}
+        projects = {p.id: p for p in Project.objects.filter(id__in=project_ids)}
+        group_ids = {event.group_id for event in _events}
+        groups = {g.id: g for g in Group.objects.filter(id__in=group_ids)}
+        for event in _events:
+            event.project = projects[event.project_id]
+            event.group = groups[event.group_id]
+        eventstore.bind_nodes(_events, "data")
+
+    events = get_events(retention_days)
+    count = events.count()
+
+    if skip_backfill:
+        print("Skipping backfill\n")
+        return
```

Comment:
This block should probably precede the `events =` and `counts =` lines above.

Reply:
Good point